src/main/java/com/conveyal/gtfs/graphql/GTFSGraphQL.java

```diff
@@ -1,41 +1,93 @@
+        DataLoader<JDBCFetcher.JdbcQuery, List<Map<String, Object>>> jdbcQueryDataLoader = newJdbcDataLoader();
+        registry.register("jdbcfetcher", jdbcQueryDataLoader);
+
+        return GraphQL.newGraphQL(GraphQLGtfsSchema.feedBasedSchema)
+            // this instrumentation implementation will dispatch all the dataloaders as each level fo the graphql query
+            // is executed and hence make batched objects available to the query and the associated DataFetchers.  This
+            // must be created new each time to avoid caching load results in future requests.
+            .instrumentation(new DataLoaderDispatcherInstrumentation(registry))
+            .build()
+            // we now have a graphQL schema with dataloaders instrumented.  Time to make a query
+            .execute(
+                // Build the execution input giving an environment context, query and variables.  The environment
+                // context is used primarily in the JDBCFetcher to get the current loader and dataSource from which
+                // the sql queries can be executed.
+                ExecutionInput.newExecutionInput()
+                    .context(new GraphQLExecutionContext(jdbcQueryDataLoader, dataSource))
+                    .query(query)
+                    .variables(variables)
+                    .build()
+            )
+            .toSpecification();
+    }
 
-    // TODO Is it correct to share one of these objects between many instances? Is it supposed to be long-lived or threadsafe?
-    // Analysis-backend creates a new GraphQL object on every request.
-    private static GraphQL GRAPHQL;
+    public static final Map<String, Object> introspectedSchema = GraphQL.newGraphQL(GraphQLGtfsSchema.feedBasedSchema)
+        .build()
+        .execute(IntrospectionQuery.INTROSPECTION_QUERY)
+        .toSpecification();
 
-    /** Username and password can be null if connecting to a local instance with host-based authentication. */
-    public static void initialize (DataSource dataSource) {
-        GTFSGraphQL.dataSource = dataSource;
-        GRAPHQL = GraphQL.newGraphQL(GraphQLGtfsSchema.feedBasedSchema)
-            .build();
+    /**
+     * Helper to obtain the jdbcQueryLoader.  It is critical that the same dataloader in the DataloaderRegistry /
+     * DataLoaderDispatcherInstrumentation be used otherwise the queries will never be able to be dispatched.
+     */
+    public static DataLoader<JDBCFetcher.JdbcQuery, List<Map<String, Object>>> getJdbcQueryDataLoaderFromContext (
+        DataFetchingEnvironment environment
+    ) {
+        return ((GraphQLExecutionContext)environment.getContext()).jdbcQueryDataLoader;
     }
 
-    public static Connection getConnection() {
-        try {
-            return dataSource.getConnection();
-        } catch (SQLException ex) {
-            throw new RuntimeException(ex);
+    /**
+     * Helper method to return the DataSource from the DataFetchingEnvironment
```

Comment:
This Javadoc could be considered trivial, as it refers only to the return and parameter types. Maybe expand on how it's used rather than the types it handles - it seems that this is used to retrieve an object representing the SQL database, to allow access to it from within a GraphQL query.